python/src/scipp/plot/plot_2d.py

```diff
@@ -251,29 +226,81 @@ def update_axes(self):
-                if self.params["masks"][self.name]["show"]:
-                    self.im[self.get_mask_key(key)].set_extent(extent_array)
-                self.ax[key].set_xlim(axparams["x"]["lims"])
-                self.ax[key].set_ylim(axparams["y"]["lims"])
-            self.ax[key].set_xlabel(axparams["x"]["labels"])
-            self.ax[key].set_ylabel(axparams["y"]["labels"])
-            for xy, param in axparams.items():
-                if self.slider_ticks[self.name][param["dim"]] is not None:
-                    getattr(self.ax[key], "set_{}ticklabels".format(xy))(
-                        self.get_custom_ticks(ax=self.ax[key],
-                                              dim=param["dim"],
-                                              xy=xy))
+
+        # Create coordinate axes for resampled array to be used as image
+        self.xyrebin["x"] = sc.Variable(
+            dims=[axparams['x']["dim"]],
+            values=np.linspace(extent_array[0], extent_array[1],
+                               self.image_resolution[0] + 1),
+            unit=self.slider_x[self.name][axparams['x']["dim"]].unit)
+        self.xyrebin["y"] = sc.Variable(
+            dims=[axparams['y']["dim"]],
+            values=np.linspace(extent_array[2], extent_array[3],
+                               self.image_resolution[1] + 1),
+            unit=self.slider_x[self.name][axparams['y']["dim"]].unit)
+        self.image_dx = self.xyrebin["x"].values[1] - self.xyrebin["x"].values[
+            0]
+        self.image_dy = self.xyrebin["y"].values[1] - self.xyrebin["y"].values[
+            0]
+
+        # Create bin-edge coordinates in the case of non bin-edges, since rebin
+        # only accepts bin edges.
+        xdims = self.xyrebin["x"].dims
+        if not self.histograms[self.name][xdims[0]]:
+            self.xyedges["x"] = sc.Variable(
+                dims=xdims,
+                values=centers_to_edges(
+                    self.slider_x[self.name][xdims[0]].values),
+                unit=self.slider_x[self.name][xdims[0]].unit)
+        else:
+            self.xyedges["x"] = self.slider_x[self.name][xdims[0]].astype(
+                sc.dtype.float64)
+        ydims = self.xyrebin["y"].dims
+        if not self.histograms[self.name][ydims[0]]:
+            self.xyedges["y"] = sc.Variable(
+                dims=ydims,
+                values=centers_to_edges(
+                    self.slider_x[self.name][ydims[0]].values),
+                unit=self.slider_x[self.name][ydims[0]].unit)
+        else:
+            self.xyedges["y"] = self.slider_x[self.name][ydims[0]].astype(
+                sc.dtype.float64)
+
+        # Compute bin widths for normalization pre-rebin in order to retain the
+        # pixel values instead of spreading the data over the pixels.
+        self.xywidth["x"] = sc.Variable(
+            dims=self.xyrebin["x"].dims,
+            values=np.ediff1d(self.xyedges["x"].values) / self.image_dx)
```

Comment:
What is the reason for using `ediff1` instead of simply `self.xyedges["x"][dim, 1:] - self.xyedges["x"][dim, :-1]`?